pystac_client/client.py

```diff
@@ -619,6 +620,144 @@ def search(
             modifier=self.modifier,
         )
 
+    def collection_search(
+        self,
+        *,
+        max_collections: Optional[int] = None,
+        limit: Optional[int] = None,
+        bbox: Optional[BBoxLike] = None,
+        datetime: Optional[DatetimeLike] = None,
+        q: Optional[str] = None,
```

Comment:
I'm 👎🏼 on a short name like this — I'd prefer

```suggestion
        free_text_query: Optional[str] = None,
```

`free_text` could also work, I reckon.

Reply:
`free_text` is fine with me but do we want to stray away from the actual query parameter name from the API? I agree that it is opaque and that something more descriptive would be nice.

Reply:
Ah, you're right about that, I missed the `q` in my quick skim of the extension readme. Let's stay with `q`, no reason to add more names to the mix.